src/main/java/com/aws/greengrass/util/S3SdkClientFactory.java

```diff
@@ -88,4 +95,30 @@ public S3Client getClientForRegion(Region r) {
                 .serviceConfiguration(S3Configuration.builder().useArnRegionEnabled(true).build())
                 .credentialsProvider(credentialsProvider).region(r).build());
     }
+
+    /**
+     * Set s3 endpoint type.
+     *
+     * @param type s3EndpointType
+     */
+    private void setS3EndpointType(String type) {
+        //Check if system property and device config are consistent
+        //If not consistent, set system property according to device config value
+        String s3EndpointSystemProp = System.getProperty(S3_ENDPOINT_PROP_NAME);
+        boolean isGlobal = S3EndpointType.GLOBAL.name().equals(type);
+
+        if (isGlobal && S3_REGIONAL_ENDPOINT_VALUE.equals(s3EndpointSystemProp)) {
+            System.clearProperty(S3_ENDPOINT_PROP_NAME);
+            refreshClientCache();
```

Comment:
does the client actually need to be recreated? I'm not so sure about that. I bet it might read the property when making the API call, so you can just change the system property without creating the client. 

Check this.

Reply:
Yes, if we don't recreate, the endpoint won't change and we have to restart Nucleus to make the endpoint update. I think the client is the one reading system property.

Reply:
Yes, I just checked. The client does need to be recreated unfortunately.